api/api/views/health_views.py

```diff
@@ -13,6 +14,10 @@ class ElasticsearchHealthcheckException(APIException):
     status_code = status.HTTP_503_SERVICE_UNAVAILABLE
 
 
+class DatabaseHealthCheckException(APIException):
```

Comment:
A small change to ask to combine these, and use a different error code to distinguish between the two. The benefit of this is also in the test, where we can actually assert the response message aligns with the expected failure. Right now, the test could absolutely just not work due to a small misconfiguration or misunderstanding of the implementation details of a particular check, and we'd never know (if another check failed), because they all just check for 503, which has at least two meanings from this view.

Bonus points if we can remove the inconsistently named ES class's "Healthcheck" (vs "HealthCheck" to match the view).

Reply:
When you say different error code, does that mean different HTTP status? If so, which one would would you recommend for each? Also d'oh, apologies about the misnaming!

Reply:
> Also d'oh, apologies about the misnaming!

You misunderstood, I think you got the naming _right_, at least in that it actually matches the view class's name. I don't know which is "correct" HealthCheck or Healthcheck (I think of it as one word but my spell checker doesn't agree :shrug:) but consistency with the view is better, I think.

> When you say different error code, does that mean different HTTP status?

No, sorry, I mean the `APIException` class's `code` attribute: https://github.com/encode/django-rest-framework/blob/master/rest_framework/exceptions.py#L106

It can be set in the constructor. So then we'd just have a generic `HealthCheckException`, and when raising it, pass a meaningful code and/or detail to the constructor:

```py
raise HealthCheckException(code='elasticsearch', detail='Yellow cluster health')

raise HealthCheckException(code='postgres', detail='Unreachable')
```

And so forth. Main thing for this PR would be to unify them and distinguish based on the code, not necessarily that the whole method should be updated to use it correctly. Up to you how far to go with it.

Reply:
Ahhh, thank you for the explanation! That makes sense, I'll go ahead with that change 🙂

Reply:
Hmm, what's interesting is that the provided `code` value doesn't show up in the response sent back to the client, only `detail` 🤔 I'll have to look into why that's the case.

Reply:
Maybe just needs to be in the detail then. `code` might genuinely just be for internal server side use.